### PR TITLE
Attempt: Backend post comment

### PR DESCRIPTION
**Pull request by dolivas644**
Attempt to set up POST request for users to add comments for specific anime page (anime_id)

### CODE REVIEW

File: server/routes/comments.js [hunk 1]
```diff
@@ -15,4 +15,27 @@ router.get('/', async (req, res) => {
       }
     })
   
+
+router.post('/', async (req, res) => {
+    const listItem = {
+      sub: req.body.user.sub,
```
Comment (dolivas644) [inline server/routes/comments.js]: cannot read properties of undefined req.body.user.sub

File: server/routes/comments.js [hunk 2]
```diff
@@ -15,4 +15,27 @@ router.get('/', async (req, res) => {
...
+    console.log(listItem);
+  
+    //select id from users where sub == sub
+    const userSub = `SELECT id FROM users WHERE sub =$1`;
+    const subInfo = await db.query(userSub, [listItem.sub])
+    console.log(subInfo, "Finds respective user id from given sub");
+    const user_id = subInfo[0].id;
+
+    //query adds info to the junction table containing user_id & anime_id
+    const query = 'INSERT INTO comment(user_id, anime_id, Text) VALUES($1, $2) RETURNING *'
```
Comment (dolivas644) [inline server/routes/comments.js]: Column Text IS ONLY being recognized as a column wwhenit has "Text"

Comment (dolivas644) [reply server/routes/comments.js]: Working Postico query INSERT INTO comment("Text", "user_id", "anime_id") VALUES($1, $2, $3)

File: server/routes/comments.js [hunk 3]
```diff
@@ -5,14 +5,46 @@ import * as dotenv from 'dotenv';
...
+    console.log(subInfo, "Finds respective user id from given sub");
+    const user_id = subInfo[0].id;
+
+    //select anime_id from animelist if it exists
+    const animeId = `SELECT mal_id FROM anime WHERE mal_id=$1 LIMIT 1`;
+    const valueAnime = [listItem.anime_id]
+    const resultsAnime = await db.query(animeId, valueAnime);
+
+    //query adds info to the junction table containing user_id & anime_id
+    const query = 'INSERT INTO comment("Text", "user_id", "anime_id") VALUES($1, $2, $3) RETURNING *'
```
Comment (dolivas644) [inline server/routes/comments.js:50]: error: INSERT has more target columns than expressions

{Text: 'ssss', anime_id: 322, user_id:5} 
Postman: {"Text": ",,,,", "user_id":5, "anime_id": 322}

File: server/routes/comments.js [hunk 4]
```diff
@@ -5,14 +5,46 @@ import * as dotenv from 'dotenv';
...
+    console.log(listItem);
+
+    //select id from users where sub == sub
+    const userSub = `SELECT id FROM users WHERE sub =$1`;
+    const subInfo = await db.query(userSub, [listItem.sub])
+    console.log(subInfo, "Finds respective user id from given sub");
+    const user_id = subInfo[0].id;
+
+    //select anime_id from animelist if it exists
+    const animeId = `SELECT mal_id FROM anime WHERE mal_id=$1 LIMIT 1`;
```
Comment (dolivas644) [inline server/routes/comments.js:38]: Possible Issue: If the user has NOT added this anime to their list there is a chance the anime is not in the database, as a possibility to combat this issye we can nsert to the anime table to grab the anime data to be able to have a valid anime_id

Comment (cdm-ium) [reply server/routes/comments.js:38]: Where is the code that inserts an anime into the anime table?

I think I agree that the best thing to do would be to lazy load anime. However, for a first pass, I would probably recommend not doing this and just throwng an error if you can't find the anime in your db. That way you can test the comments feature end-to-end and then go back and fix this edge case.

File: server/routes/comments.js (hunk 3 above)
Comment (cdm-ium) [reply server/routes/comments.js:50]: I do see some issues with your query. The columns that you list need to match the order that values are supplied on line 44. I'm not sure if that's the only problem though.
